playground/ExcalidrawWrapper.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from "react";
+import {
+  Excalidraw,
+  convertToExcalidrawElements,
+} from "@excalidraw/excalidraw";
+import type { ExcalidrawImperativeAPI } from "@excalidraw/excalidraw/types/types.js";
+import { parseMermaid } from "../src/parseMermaid";
+import { graphToExcalidraw } from "../src/graphToExcalidraw";
+import { DEFAULT_FONT_SIZE } from "../src/constants";
+import type { MermaidData } from "./";
+
+interface ExcalidrawWrapperProps {
+  mermaidDefinition: MermaidData["definition"];
+  onMermaidDataParsed: (
+    mermaid: MermaidData["output"],
+    error?: unknown
+  ) => void;
+}
+const ExcalidrawWrapper = ({
+  mermaidDefinition,
+  onMermaidDataParsed,
+}: ExcalidrawWrapperProps) => {
+  const [excalidrawAPI, setExcalidrawAPI] =
+    useState<ExcalidrawImperativeAPI | null>(null);
+
+  useEffect(() => {
+    if (!excalidrawAPI) {
+      return;
+    }
+
+    if (mermaidDefinition === "") {
+      excalidrawAPI.resetScene();
+      return;
+    }
+
+    const updateDiagram = async () => {
+      try {
+        const mermaid = await parseMermaid(mermaidDefinition);
+
+        const { elements, files } = graphToExcalidraw(mermaid, {
+          fontSize: DEFAULT_FONT_SIZE,
+        });
+
+        onMermaidDataParsed(mermaid);
```

Comment:
This is something not related to this excalidraw wrapper, so it's better we move to `index.tsx`, that way the flow will be clear.
Lets keep the code only related to Excalidraw updates here

Reply:
oh nice catch! It got much better after I applied your suggestion to keep the same signature.
done in [192f306](https://github.com/excalidraw/mermaid-to-excalidraw/pull/52/commits/192f30684dcb2c81bffe4dfb80d0b99b7e9399e3)